### PR TITLE
add Docker related file

### PR DESCRIPTION
**Pull request by macauleycheng**

### CODE REVIEW

**Comment by utoni** [comment]
I am interested in a working `Dockerfile`. :heart:

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+
+FROM ubuntu:22.10
+WORKDIR /root
+RUN apt-get -y update && apt-get -y install libpcap-dev
+
+COPY --from=builder /root/nDPId/libnDPI/ /root/
+COPY --from=builder /root/nDPId/build/nDPIsrvd /root/nDPId/build/nDPId /root/
+
+#RUN echo "#!/bin/bash\n" \
```
Comment (utoni) [inline Dockerfile:21]: Why commenting this out?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#for dev, uncomment below
+#RUN mkdir /root/nDPId
+#COPY . /root/nDPId/
+
+RUN cd nDPId && mkdir build && cd build && cmake .. -DBUILD_NDPI=ON && make
+
+
+FROM ubuntu:22.10
+WORKDIR /root
+RUN apt-get -y update && apt-get -y install libpcap-dev
```
Comment (utoni) [inline Dockerfile:16]:
```suggestion
RUN apt-get -y update && apt-get -y install libpcap
```

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,25 @@
+FROM ubuntu:22.10 as builder
+
+WORKDIR /root
+RUN apt-get -y update && apt-get install -y git cmake pkg-config libpcap-dev autoconf libtool
+
+RUN git clone https://github.com/utoni/nDPId.git
+#for dev, uncomment below
```
Comment (utoni) [inline Dockerfile:7]: I do not understand this comment.

File: Dockerfile (hunk 1 above)
Comment (macauleycheng) [reply Dockerfile:21]: Before push code to github, the commnet line is used for local build and test.
After push to github, we can the github code to build.

File: Dockerfile (hunk 2 above)
Comment (macauleycheng) [reply Dockerfile:16]: it shall be libpcap0.8, libpcap can't be found

**Review by utoni — APPROVED** [review]
Since I am not that familiar with docker, I think it can be merged and changed later (if necessary).

**Comment by stefanDeveloper** [comment]
I created a Docker Image for nDPId with consumer and producer split:
https://github.com/stefanDeveloper/heiDPI

**Pull request merged**